src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -285,12 +285,16 @@ private void OnException(HttpContext httpContext, Exception exception)
                 var telemetry = httpContext?.Features.Get<RequestTelemetry>();
                 if (telemetry != null)
                 {
-                    telemetry.Success = false;
+                    if (int.TryParse(telemetry.ResponseCode, out int _responseCode))
+                    {
+                        telemetry.Success = !(_responseCode == 200);
```

Comment:
After reviewing the original issue and this PR more closely I believe that this logic is a bug and that the original issue in #192 no longer existed.